### PR TITLE
Nav bar

### PR DESCRIPTION
**Pull request by ZahraSucre**
navbar visually completed but signup/login not functional yet.

### CODE REVIEW

File: frontend/src/components/NavBar.css [hunk 1]
```diff
@@ -0,0 +1,130 @@
+/* Reset styles */
+*,
+*::before,
+*::after {
+  margin: 0;
+  padding: 0;
+  box-sizing: border-box;
+}
+
+.btn-12,
```
Comment (pablisch) [inline frontend/src/components/NavBar.css]: btn-12 is not descriptive of the function or the element. Could this be something like 'nav-btn'?
This applies to all mention of btn-12. Quick to change with find and replace.

File: frontend/src/components/NavBar.css [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+}
+
+.btn-12 {
+  -webkit-tap-highlight-color: transparent;
+  -webkit-appearance: button;
+  background-color: #000;
+  background-image: none;
+  color: #fff;
+  cursor: pointer;
+  font-family: ui-sans-serif, system-ui, -apple-system, BlinkMacSystemFont,
```
Comment (pablisch) [inline frontend/src/components/NavBar.css]: There are a lot of fonts here and perhaps none are needed as Harry has set up a global font-family. If you want a specific font (possibly to import) that would do and a backup family. Happy to discuss any of the above.

File: frontend/src/components/NavBar.css [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+    padding: 1rem 3rem;
+  }
+/* Updated styles for the link container */
+.nav {
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+}
+
+.linkcontainer {
```
Comment (pablisch) [inline frontend/src/components/NavBar.css]: Similar to 'btn-12' could this be descriptive of its location, e.g. nav-link-container?

File: frontend/src/components/NavBar.jsx [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+import React from 'react';
+import { Link, useMatch, useResolvedPath } from 'react-router-dom';
+import './NavBar.css';
+
+const NavBar = ({ navigate }) => {
+    const logout = () => {
+      window.localStorage.clear();
+      window.localStorage.removeItem('token');
+      window.localStorage.removeItem('token');
```
Comment (pablisch) [inline frontend/src/components/NavBar.jsx]: There are 3 lines here that seem to perform the same purpose. I am assuming that clear() will remove everything including the token.

File: frontend/src/components/NavBar.jsx [hunk 5]
```diff
@@ -0,0 +1,38 @@
+// NavBar.js
+
+import React from 'react';
+import { Link, useMatch, useResolvedPath } from 'react-router-dom';
```
Comment (pablisch) [inline frontend/src/components/NavBar.jsx]: None of these appear to be used. Fine to leave them if they have some future purpose.
Link is imported but navigate is used. I am not sure but have a feeling that you will be using wither one or the other.

**Review by pablisch — CHANGES_REQUESTED** [review]
Hey Zahra. Nice work. A few minor points on class labelling so that it is clear to anyone that comes to the code later and what looks like some excess code presumably from boiler-plates. Let me know if it is helpful to discuss any of this.

File: frontend/src/components/NavBar.jsx
Comment (ChalkandFeather) [inline frontend/src/components/NavBar.jsx:1]: Recommend label "btn-12 baldbible-button" should reflect "Baldbible Home-button"

The 'Baldbible button' would really benefit from a meaningful Aria Label to assist screen readers to announce the context of the button.

`<button className="btn-12 baldbible-button" onClick={() => navigate("/")} aria-label="Navigate to Baldbible">
  <span>Baldbible</span>
</button>`

Can also this for SignUp and Log in buttons too
SignUP
`<button className="btn-12" onClick={() => navigate("/SignUp")} aria-label="Navigate to Sign Up">
  <span>SignUp</span>
</button>`
SignUp

<button className="btn-12" onClick={() => navigate("/Login")} aria-label="Navigate to Login">
  <span>Login</span>
</button>

**Review by ChalkandFeather — CHANGES_REQUESTED** [review]
Hi there, I've added some aria-labelling that can be added to your existing code for the buttons.

File: frontend/src/components/NavBar.css (hunk 2 above)
Comment (hjtrhodes) [reply frontend/src/components/NavBar.css]: I actually removed the global font family but it is worth setting it up again, takes 2 minutes

**Pull request merged**